src/Controller/ParticipantDetailsController.php

```diff
@@ -231,4 +262,56 @@ public function participantConsent(
         $response->headers->set('Content-Type', 'application/pdf');
         return $response;
     }
+
+    /**
+     * @Route("/participant/{id}/incentive/{incentiveId}", name="participant_incentive", defaults={"incentiveId": null})
+     */
+    public function participantIncentiveAction(
+        $id,
+        $incentiveId,
+        Request $request,
+        LoggerService $loggerService,
+        EntityManagerInterface $em,
+        ParticipantSummaryService $participantSummaryService,
+        SiteService $siteService
+    ): Response {
+        $participant = $participantSummaryService->getParticipantById($id);
+        $incentive = $incentiveId ? $em->getRepository(Incentive::class)->find($incentiveId) : null;
+        $incentiveForm = $this->createForm(IncentiveType::class, $incentive, ['action' => $request->getRequestUri(), 'require_notes' => true]);
+        $incentiveForm->handleRequest($request);
+        if ($incentiveForm->isSubmitted()) {
+            if ($incentiveForm->isValid()) {
+                $userRepository = $em->getRepository(User::class);
+                $now = new \DateTime();
+                $incentive = $incentiveForm->getData();
+                $incentive->setParticipantId($id);
+                $user = $userRepository->find($this->getUser()->getId());
```

Comment:
Do we not already have a method to get the current user's user entity? I remember that this isn't as straightforward as it should be because the token user isn't the user entity, but we should probably centralize this somewhere.